hrjobcontract/js/test/controllers/contract_test.js

```diff
@@ -53,6 +54,43 @@ define([
       it('retrieves contract full details', function () {
         expect(ContractService.fullDetails).toHaveBeenCalled();
       });
+
+      describe('when loads the leave data', function () {
+        var leave = _.sample(contractMock.contractLeaves.values);
+
+        beforeEach(function () {
+          $scope.leave = [{ leave_type: leave.leave_type }];
+          $scope.revisionCurrent.jobcontract_id = 1;
+          $scope.$emit('updateContractView');
+          $scope.$digest();
+          $httpBackend.flush();
+        });
+
+        it('maps Absence Types and Job Contract data correctly', function () {
```

Comment:
shouldn't you cover all the edge cases:

* when the absence types order don't match the contract data order
* when there are more absence types than contract data (ie some absence type had been added afterwards)
* when there are less absence types than contract data (ie some absence type had been removed/disabled afterwards)

Reply:
Done.